Fixes das issues #4 e #5

joaojvds opened the pull request:
-- Mensagem de erro ao digitar um login incorreto 
-- Redirect client-side ao fazer login
-- Mensagem de confirmação do email ao cadastrar-se
-- Mensagem de erro ao inserir um email já cadastrado
-- Fixes #4 e #5

joaojvds committed:
Mensagem de erro da pagina login

--Adicionado mensagem de erro ao digitar um login incorreto

joaojvds committed:
Consertado pagina de login

--Consertado método de login
--A pagina agora faz o redirect usando a url que o server manda

GCastilho commented on client/public/index.html:
```diff
@@ -14,7 +14,6 @@
         <title>React App</title>
     </head>
     <body>
-        <noscript>You need to enable JavaScript to run this app.</noscript>
```
Pq tirar essa linha?

joaojvds replied:
Percebi que era melhor mesmo deixar essa linha, então coloquei de volta

joaojvds committed:
Handles da pagina de cadastro

--Adicionado handle de erros na pagina de cadastro
--Criado pagina pagina de confirmação de email (funções da pagina ainda a serem implementadas )
--Mudanças no css para remover o flicker ao trocar de pagina

joaojvds committed:
Fix unnecessary page reload

--Trocado funções que fazem redirect do html para as do react
--Removido arquivo app

GCastilho commented on client/src/pages/login/Login.css:
```diff
@@ -13,4 +14,15 @@
+    border: 1px solid crimson;
+    border-radius: 5px;
+    background-color: palevioletred;
+    color: crimson;
```
o contraste entre a background color e a cor do texto tá muito baixo (é meio difícil de ler), dá uma melhorada nisso

joaojvds replied:
Mudei as tonalidades de vermelho da mensagem de erro, agora tem um melhor contraste

GCastilho commented on client/src/pages/register/Register.jsx:
```diff
@@ -1,21 +1,79 @@
+            this.setState({errorMsg: 'A senha deve conter pelo menos uma letra minuscula'});
+        } else if (this.state.password.search(/[A-Z]/) === -1) {
+            this.setState({errorMsg: 'A senha deve conter pelo menos uma letra maiucula'});
+        }*/ else {
```
Tira esses checks não usados, se/quando formos fazer esse tipo de check, o servidor irá fazê-lo e retornar a errorMsg adequada (o cliente apenas irá inseri-la na tela), não sendo necessário ter esse check no cliente, ou esse código pra adicionar depois (pq não vai adicionar anyway)

joaojvds replied:
Apaguei os checks de senha da pagina

GCastilho commented on client/src/pages/register/Register.jsx:
```diff
@@ -1,21 +1,79 @@
+                        <RoundButton label='Cadastrar' onClick={this.handleSubmit} type='submit'/>
+                    </form> :
+                    <div className='confirmWindow'>
+                        <h1>Quase lá</h1>
```
Troca essa mensagem para algo mais "formal" tipo "Confirme o email" ou "necessário confirmar/confirmação de email", algo do tipo, mas tira esse "quase lá" daí, pq além de estranho ele faz parecer que a conta não foi _criada_, o que ela foi

GCastilho commented on client/src/pages/register/Register.jsx:
```diff
@@ -1,21 +1,79 @@
+                    </form> :
+                    <div className='confirmWindow'>
+                        <h1>Quase lá</h1>
+                        <p>O email <b>{this.state.email}</b> esta correto ?</p>
```
E troca essa mensagem pra uma msg mais adequada tbm, tipo "enviamos um email de confirmação de cadastro para <email>, para ativar sua conta por favor siga as instruções informadas no email" ou algo disso tipo (não precis ser _essa_ mensagem nem nada, é uma sugestão)

GCastilho reviewed (changes requested):
Algumas modificações pontuais no código descritas nominalmente e um detalhe relevante:

É necessário atualizar o react-script por uma vulnerabilidade, então rode um 'npm audit fix' ('npm audit' pra detalhes) no cliente/frontend/react para ele atualizar as dependências dos arquivos package.json e é pra estar tudo certo

joaojvds committed:
Requested changes

--Mudado esquema de cores das mensagens de erro
--Removido checks do password
--Alterado mensagem de confirmação de email para algo mais formal

joaojvds committed:
Alterado carregamento da pagina

--Mudado forma de carregamento para um carregamento dinâmico

GCastilho committed:
Adicionado fallback de Loading

-- Usar o lazy loading sem o Suspense faz o app dar um flicker, apagando tudo na tela até o loading ter terminado (mais visível em internet lenta), isso é resolvido ao dicionar Suspense no router com fallback com uma mensagem de 'Loading' (ou qualquer outra) até o carregamento ser terminado

joaojvds committed:
Adicionado tela de loading

--Adicionado um loading ao trocar de pagina

GCastilho commented on client/src/index.js:
```diff
@@ -2,16 +2,14 @@ import React from 'react';
-const Header = React.lazy(() => import('./components/Header/Header'));
-const Routes = React.lazy(() => import('./routes'));
+import Header from './components/Header/Header';
+import Routes from './routes';
```
Pq mesmo vc voltou pra o que tava antes?

joaojvds replied:
pq essa pagina é carregada junto com o react e não faz diferença no tempo de carregamento

joaojvds committed:
Alterado icon do loading

--Alterado icon de 'bubbles' para 'spinningBubbles'

GCastilho reviewed (approved):
Digamos que ficou realmente bom e mais otimizado que antes, parabéns
